File: Biblioteca/src/main/java/com/biblioteca/service/LibroService.java
package com.biblioteca.service;

import java.util.List;

import com.biblioteca.domain.libro.Libro;

public class LibroService {

        
    public LibroService() {
    }

    public boolean available(Libro libro) {
        return libro.getStock() > 0;
    }

    public Libro getLibroByTitulo(List<Libro> libros, String tituloLibro) {
        return libros.stream()
                 .filter(libro -> libro.getTitulo().equalsIgnoreCase(tituloLibro))
                 .findFirst()
                 .orElse(null);
    }
    
}

File: python/biblioteca/src/service/libro_service.py
from domain.libro.libro import Libro
class LibroService:
    def __init__(self):
        pass

    def get_libro_by_nombre(self, libros: list[Libro], nombre) -> Libro:
        encontrado = [l for l in libros if l.nombre == nombre]
        return encontrado[0] if len(encontrado) == 1 else None
    